src/services/PersonInfoRequestProcessor.ts

```diff
@@ -0,0 +1,78 @@
+import { Metrics } from "@aws-lambda-powertools/metrics";
+import { Logger } from "@aws-lambda-powertools/logger";
+import NodeRSA from "node-rsa";
+import { BavService } from "./BavService";
+import { HttpCodesEnum } from "../models/enums/HttpCodesEnum";
+import { MessageCodes } from "../models/enums/MessageCodes";
+import { EnvironmentVariables } from "../utils/Constants";
+import { createDynamoDbClient } from "../utils/DynamoDBFactory";
+import { checkEnvironmentVariable } from "../utils/EnvironmentVariables";
+import { getFullName } from "../utils/PersonIdentityUtils";
+import { Response } from "../utils/Response";
+
+export class PersonInfoRequestProcessor {
+  private static instance: PersonInfoRequestProcessor;
+
+  private readonly logger: Logger;
+
+  private readonly metrics: Metrics;
+
+  private readonly BavService: BavService;
+
+	private readonly personIdentityTableName: string;
+
+	private readonly publicKey: string;
+
+	constructor(logger: Logger, metrics: Metrics, publicKey: string) {
+		this.publicKey = publicKey;
+		this.personIdentityTableName = checkEnvironmentVariable(EnvironmentVariables.PERSON_IDENTITY_TABLE_NAME, logger);
+  	const sessionTableName: string = checkEnvironmentVariable(EnvironmentVariables.SESSION_TABLE, logger);
+
+  	this.logger = logger;
+  	this.metrics = metrics;
+  	this.BavService = BavService.getInstance(sessionTableName, this.logger, createDynamoDbClient());
+	}
+
+	static getInstance(logger: Logger, metrics: Metrics, publicKey: string): PersonInfoRequestProcessor {
+  	if (!PersonInfoRequestProcessor.instance) {
+  		PersonInfoRequestProcessor.instance = new PersonInfoRequestProcessor(logger, metrics, publicKey);
+  	}
+  	return PersonInfoRequestProcessor.instance;
+	}
+
+	async processRequest(sessionId: string): Promise<Response> {
+  	const person = await this.BavService.getPersonIdentityById(sessionId, this.personIdentityTableName);
+  	const session = await this.BavService.getSessionById(sessionId);
+
+		if (!person) {
+  		this.logger.error("No person found for session id", {
+  			messageCode: MessageCodes.PERSON_NOT_FOUND,
+  		});
+  		return new Response(HttpCodesEnum.UNAUTHORIZED, `No person found with the session id: ${sessionId}`);
+  	}
+
+  	if (!session) {
```

Comment:
[nit] do you think it's worth putting to session check 1st maybe? Just so if session exists but person doesn't we have govuk_signin_journey_id in logs to lookup, having person 1st doesn't give us any obvious benefits

Reply:
Shame we're having to query session DB just got the govuk_signin_journey_id though :/

Reply:
Swapped!